.generator/templates/model/ObjectSerializer.mustache

```diff
@@ -197,14 +199,15 @@ export class ObjectSerializer {
                     try {
                         oneOfs.push(ObjectSerializer.deserialize(data, oneOf, format))
                     } catch (e) {
-                        console.debug(`could not deserialize ${oneOf} (${e})`)
+                        if (e instanceof ParseException) {
+                            if (e.type == "enum") {
+                                continue;
+                            }
+                        }
                     }
                 }
-                if (oneOfs.length > 1) {
-                    throw new TypeError(`${data} matches multiple types from ${oneOfMap[type]} ${oneOfs}`);
-                }
-                if (oneOfs.length == 0) {
-                    throw new TypeError(`${data} doesn't match any type from ${oneOfMap[type]} ${oneOfs}`);
+                if (oneOfs.length != 1) {
+                    throw new ParseException("oneOf", "Unrecognized oneOf");
```

Comment:
You changed the exception, but you're never doing anything with it.

Reply:
To be resilient, we don't actually want to throw on oneOf deserialisation, we want to get a generic object which contains the raw data.
To do that, you might want to create a class `UnparsedObject`, which you would add as an element of every oneOf union type, so that an object of this class is always a valid type of any oneOf type